packages/babel-core/src/transformation/file/index.js

```diff
@@ -302,7 +302,9 @@ export default class File extends Store {
       ref._generated = true;
       ref.id = uid;
       ref.type = "FunctionDeclaration";
-      this.path.unshiftContainer("body", ref);
+      this.path
+        .unshiftContainer("body", ref)
+        .forEach(path => this.path.requeue(path));
```

Comment:
Won't this cause fun things such as requiring the typeof helper to implement the typeof helper?